### DISCUSSION TITLE
how to fix key warning when I use loading.tsx with ppr in Next.js App Router

### DISCUSSION THREAD

**Discussion by rehatch-yoshimori-otsuka** [vercel/next.js#68430] [Help] [answered]
### Summary

- combination of loading.tsx and ppr true makes blow key warning.
  - `Each child in a list should have a unique "key" prop. See https://react.dev/link/warning-keys for more information.`
- disappear the warning when I rename loading.tsx to 'loading-dummy.tsx' or set ppr false.

https://github.com/user-attachments/assets/5407203c-d893-4cd8-85cc-9ab8bed48f94

`src/app/loading.tsx`
```tsx
// TODO: how to fix key warning when I use loading.tsx with ppr in Next.js App Router
// HERE!! rename loading-dummy.tsx to ignore key warning
const Loading = () => {
  return <div>Loading...</div>;
}

export default Loading
```

next.config.mjs
```mjs
/** @type {import('next').NextConfig} */
const nextConfig = {
    experimental: {
        // TODO: how to fix key warning when I use loading.tsx with ppr in Next.js App Router
        // HERE!! set ppr false to ignore key warning
        ppr: true,
    }
};

export default nextConfig;
```

### Additional information

```js
## sandbox

[error-loading.zip](https://github.com/user-attachments/files/16467430/error-loading.zip)


- pnpm 8.15.8
- nodejs 20.16.0
- next 15.0.0-canary.98
- react 19.0.0-rc-06d0b89e-20240801
```

### Example

https://github.com/user-attachments/files/16467430/error-loading.zip

**Comment by rehatch-yoshimori-otsuka** [comment]
Is this an Issue?

**Comment by DenisBessa** [comment, marked as answer]
The problem seems to not occur when `loading.tsx` has a `"use client"` directive.

**Reply by rehatch-yoshimori-otsuka** [reply]
Thank you very much!
I can check your advice.
You can close this discussion and you can also answer [this issue](https://github.com/vercel/next.js/issues/68634) if the solutions are same.

for remind, I'll show you the fixed code.

the fixed code doesn't occur the error `Each child in a list should have a unique "key" prop. See https://react.dev/link/warning-keys for more information.` any more.

app/loading.tsx
```tsx
'use client' // <- can FIX below TODO

// TODO: how to fix key warning when I use loading.tsx with ppr in Next.js App Router
// HERE!! rename loading-dummy.tsx to ignore key warning
const Loading = () => {
  return <div>Loading...</div>;
}

export default Loading
```

**Reply by jperezrealini** [reply]
But the point of having a `loading.tsx` is that it's statically rendered. Making it a client component defeats that purpose.

**Reply by khuezy** [reply]
Agreed, that isn't a fix.

**Comment by gaojude** [comment]
Cannot produce this issue in 15.0.3-canary.2 so it's fixed in the latest canary.
